protocol/storage/deals_awaiting_seal.go

```diff
@@ -0,0 +1,117 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/filecoin-project/go-filecoin/proofs/sectorbuilder"
+	"github.com/ipfs/go-cid"
+	cbor "github.com/ipfs/go-ipld-cbor"
+)
+
+func init() {
+	cbor.RegisterCborType(dealsAwaitingSeal{})
+}
+
+// sectorInfo combines sector Metadata from rust proofs with go-filecoin specific data
+type sectorInfo struct {
+	Metadata         *sectorbuilder.SealedSectorMetadata
+	CommitMessageCid *cid.Cid
+	Succeeded        bool
+	ErrorMessage     string
+}
+
+// dealsAwaitingSeal is a container for keeping track of which sectors have
+// pieces from which deals. We need it to accommodate a race condition where
+// a sector commit message is added to chain before we can add the sector/deal
+// book-keeping. It effectively caches success and failure results for sectors
+// for tardy add() calls.
+type dealsAwaitingSeal struct {
+	l sync.Mutex
```

Comment:
nit: can we call this dealsMu (I like [hats](https://dmitri.shuralyov.com/idiomatic-go/entries/2))